### ISSUE TITLE
getText() for `type foo = string[][]` (array of string arrays) is invalid

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#13309]
Environment Info:

Node version: v14.1.0
npm version: v6.14.5
Local ESLint version: v7.0.0 (Currently used)
Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**

`babel-eslint`

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
type X = string[][]

```

**What did you expect to happen?**

Expected that `context.getSourceCode().getText(elementTypeNode))` will return `string[][]`.

**What actually happened? Please include the actual, raw output from ESLint.**

Returns `Array<string>[]`.

**Comment by anikethsaha** [comment]
Thanks for the issue 👍 

I haven't check for `babel-eslint` , I checked `context.getSourceCode().getText` in master, it seems working. 

 _didnt use the given example as default parser wont parse it._

**Comment by g-plane** [comment]
Hi @gajus , can you tell us what problems have you encountered?

**Comment by kaicataldo** [comment]
@gajus Since the default parser isn't able to parse this, is there any chance you can make an example repo we can look at? These kinds of issues that crop up are really hard to debug without a repro case because of integration between different projects.
